src/store/mainpage_reducer.js

```diff
@@ -0,0 +1,110 @@
+import axios from 'axios';
+
+const BASE_URL = 'http://127.0.0.1:3000/api/v1';
+
+// action creators
+
+const ALL_CARS = 'ALL_CARS';
+const SINGLE_CAR = 'SINGLE_CAR';
+const ADD_CAR = 'ADD_CAR';
+const DELETE_CAR = 'DELETE_CAR';
+
+// Initial state
+
+const initialState = {
+  cars: [],
+  car: {},
+};
+
+// get data from api
+
+export const fetchCars = async () => {
+  const response = await axios.get(`${BASE_URL}/all_cars`);
+  return response.data;
+};
+
+export const fetchCarDetails = async (id) => {
+  const response = await axios.get(`${BASE_URL}/show_car/${id}`);
+  return response.data;
+};
+
+export const addCar = async ({
+  name, description, price, image, userid,
+}) => {
+  try {
+    await axios.post(`${BASE_URL}/add_car`, {
+      name,
+      description,
+      owner: userid,
+      cost: price,
+      image_url: image,
+    });
+  } catch (error) {
+    console.log(error);
```

Comment:
- [ ] Kindly delete this old piece of debugging console.log code, to improve the cleanliness of the repo. 👍